AdvancedChangedFiles/SelfIntroduction.jsx

```diff
@@ -0,0 +1,161 @@
+﻿import React, { Component } from 'react';
+
+export default class SelfIntroduction extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            summary: props.summary || '',
+            description: props.description || '',
+            originalSummary: props.summary || '',
+            originalDescription: props.description || '',
+            summaryError: '',
+            descriptionError: '',
+            isEditing: false,
+        };
+
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.save = this.save.bind(this);
+        this.cancel = this.cancel.bind(this);
+        this.toggleEdit = this.toggleEdit.bind(this);
+    }
+
+    handleInputChange(event) {
+        const { name, value } = event.target;
+        this.setState({ [name]: value });
+
+        // Validation
+        if (name === 'summary') {
+            if (value.length > 150) {
+                this.setState({ summaryError: 'Summary must be no more than 150 characters.' });
+            } else {
+                this.setState({ summaryError: '' });
+            }
+        }
+
+        if (name === 'description') {
+            if (value.length < 150 || value.length > 600) {
+                this.setState({ descriptionError: 'Description must be between 150-600 characters.' });
+            } else {
+                this.setState({ descriptionError: '' });
+            }
```

Comment:
If you only have a single goal in your condition, you can use ternary operator instead of if/else. For this one you can use something like
**const message = condition ? value : other value** and then use that message in your state